subworkflows/local/decontamination.nf

```diff
@@ -22,7 +22,7 @@ workflow DECONTAMINATION {
 
     ALIGNMENT( to_align, false )
 
-    SAMTOOLS_BAM2FQ( ALIGNMENT.out.bam.map { meta, ref_fasta, bam, bai -> [ meta, bam ] }, true )
+    SAMTOOLS_BAM2FQ( ALIGNMENT.out.bam.map { meta, ref_fasta, bam, bai -> [ meta, bam ] }, reads.map { meta, reads -> meta.single_end == false } )
```

Comment:
this is the later comment I meant.. having to do `reads.map { meta, reads -> meta.single_end == false }` feels more difficult than it needs to be, unless if I misunderstand something about how the `SAMTOOLS_BAM2FQ` process is called